src/p2G4_func_queue.c

```diff
@@ -6,26 +6,34 @@
 #include "p2G4_func_queue.h"
 #include "bs_oswrap.h"
 #include "bs_tracing.h"
+#include <stdlib.h>
 
 /*
  * Array with one element per device interface
  * Each interface can have 1 function pending
  */
 static fq_element_t *f_queue = NULL;
+static uint32_t *device_list = NULL;
 
-static uint32_t next_d = 0;
+static uint32_t next_idx = 0;
 static uint32_t n_devs = 0;
+static bool flush_pend = false;
+static bool force_sort = false;
```

Comment:
(Just making sure) we don't need concurrent access protection for these because `main()` is a single thread of `while (...) { }` loop that runs until the simulation ends.

Reply:
yes currently everything is a single threaded super-loop. Concurrent access is not currently an issue.